### PR TITLE
Add fetch messages behind a toggle & some advanced settings

### PR DESCRIPTION
**Pull request by cammellos**
This PR does a few things:

1) Add fetch messages implementation on any kind of chat. It's behind a
   toggle (on by default) as design is still unsure whether we want it,
   but it's very useful for debugging.
2) Allow setting light client from mobile

It also partially remove node config management from the clojure part, as it's better if that's not explicitly managed by clients. Some parts are still relying on it but they are not functional (keycard), while others are still using it and will need to be updated eventually (syncing), in order to get rid completely of node config.

https://github.com/status-im/status-go/compare/8a4c2d8d...8cf9986a

Happy to split if necessary as that's multiple actual features, also it might use outdated conventions etc, let me know if that's the case.

### Testing

Just smoke testing to make sure basics are not broken, fetch-messages can be tested but it's a bit difficult, since you would have to replicate message loss, and then fetch from a chat.

### CODE REVIEW

File: src/status_im/waku/core.cljs [hunk 1]
```diff
@@ -131,3 +123,17 @@
...
             (navigation/navigate-back)))
+
+(rf/defn toggle-light-client
+  {:events [:wakuv2.ui/toggle-light-client]}
+  [{:keys [db]} enabled?]
+  {:db            (assoc-in db [:profile/profile :wakuv2-config :LightClient] enabled?)
+
+   :json-rpc/call [{:method     "wakuext_setLightClient"
+                    :params     [{:enabled enabled?}]
+                    :on-success #(do
```
Comment (ilmotta) [inline src/status_im/waku/core.cljs]: https://guide.clojure.style/#no-multiple-forms-fn-literals

File: src/status_im2/contexts/chat/events.cljs [hunk 2]
```diff
@@ -449,3 +449,12 @@
   {:events [:chat/check-last-chat]}
   [{:keys [db]}]
   {:chat/open-last-chat (get-in db [:profile/profile :key-uid])})
+
+(rf/defn fetch-messages
+  {:events [:chat/fetch-messages]}
```
Comment (ilmotta) [inline src/status_im2/contexts/chat/events.cljs:439]: The new endpoint syncs data for the last 30 days, so the word `sync` seems stronger to me than just `fetch`.

**Review by ilmotta — APPROVED** [review]
> It's behind a toggle (on by default) as design is still unsure whether we want it

I'm not aware of what this sync feature is, and I couldn't find any issue in status-mobile or status-go. Here's the status-go PR because it's not linked to the mobile one https://github.com/status-im/status-mobile/pull/18007.

Do you have any source describing what is this feature, what are our intentions, requirements, how it could evolve, and so on? These things would've helped me review the PR.

The code LGTM, I just don't understand where we are going with it.

> [...] also it might use outdated conventions etc, let me know if that's the case.

Side note: Some parts of the PR are touching old stuff like advanced settings, that part is too off in terms of guidelines, so I'm not commenting anything about that.

In terms of event handlers, we have agreed to avoid mixing in the same file events using `rf/reg-event-fx` and `rf/defn`, so it's okay that in this PR you're going with the flow of `rf/defn` :)

Edit: formatting

File: src/status_im2/contexts/chat/events.cljs (hunk 2 above)
Comment (cammellos) [reply src/status_im2/contexts/chat/events.cljs:439]: I went for `FetchMessages` everywhere to mirror the UI and be consistent with desktop, hope that's ok

**Comment by cammellos** [comment]
> > It's behind a toggle (on by default) as design is still unsure whether we want it
> 
> I'm not aware of what this sync feature is, and I couldn't find any issue in status-mobile or status-go. Here's the status-go PR because it's not linked to the mobile one #18007.
> 
> Do you have any source describing what is this feature, what are our intentions, requirements, how it could evolve, and so on? These things would've helped me review the PR.
> 
> The code LGTM, I just don't understand where we are going with it.
> 
> > [...] also it might use outdated conventions etc, let me know if that's the case.
> 
> Side note: Some parts of the PR are touching old stuff like advanced settings, that part is too off in terms of guidelines, so I'm not commenting anything about that.
> 
> In terms of event handlers, we have agreed to avoid mixing in the same file events using `rf/reg-event-fx` and `rf/defn`, so it's okay that in this PR you're going with the flow of `rf/defn` :)
> 
> Edit: formatting

@ilmotta there's not much documentation, I think product wants to ditch the feature alltogether, but I spoke to John and agreed that is fine to add this (disabled on release when we release). This is helpful for debugging and at least side step some of the issues with network reliability and make dogfooding a bit smoother (it's a patch of course and not a fix).

**Comment by VolodLytvynenko** [comment]
Hi @cammellos could you please resolve conflicts in the current PR? Thanx!

**Comment by cammellos** [comment]
@VolodLytvynenko I believe this has been merged alreayd in a different PR, I will close it, thank you

**Pull request closed without merging**